Sources/_StringProcessing/Regex/Match.swift

```diff
@@ -180,20 +180,13 @@ extension BidirectionalCollection where SubSequence == Substring {
 }
 
 @available(SwiftStdlib 5.7, *)
-extension Regex {
-  public init(quoting string: String) {
-    self.init(node: .quotedLiteral(string))
-  }
-}
-
-@available(SwiftStdlib 5.7, *)
-public func ~=<Output>(regex: Regex<Output>, input: String) -> Bool {
-  guard let _ = try? regex.wholeMatch(in: input) else { return false }
-  return true
+@_disfavoredOverload // disambiguate from equatable patterns
+public func ~=<R: RegexComponent>(regex: R, input: String) -> Bool {
+  input.wholeMatch(of: regex) != nil
 }
 
 @available(SwiftStdlib 5.7, *)
-public func ~=<Output>(regex: Regex<Output>, input: Substring) -> Bool {
-  guard let _ = try? regex.wholeMatch(in: input) else { return false }
-  return true
+@_disfavoredOverload // disambiguate from equatable patterns
```

Comment:
I think it would make sense to make them both static func in `extension RegexComponent`. Why did you add `@_disfavoredOverload`?

Reply:
When there's no `@_disfavoredOverload`, some tests failed as string patterns had two ambiguous overloads.  When I moved the operator inside `extension RegexComponent`, I no longer see those errors so that's good. Is that the effect of making an operator a static method?

Reply:
Yes, there is some logic we have to optimize that in the solver. If you know exactly what used to fail it would be great to reduce and add it to our test suite to make sure that doesn't regress in the future.

Reply:
In general, if possible, it's preferred to add protocol related operators into extensions of protocols since it helps to group and filter overloads in the solver.

Reply:
```swift
@available(SwiftStdlib 5.7, *)
public func ~=<R: RegexComponent>(regex: R, input: String) -> Bool {
  input.wholeMatch(of: regex) != nil
}

@available(SwiftStdlib 5.7, *)
public func ~=<R: RegexComponent>(regex: R, input: Substring) -> Bool {
  input.wholeMatch(of: regex) != nil
}

func testSwitches() {
  switch "abcde" {
  case "abc": return // error: Ambiguous use of operator '~='
  }
}
```

Reply:
Hm, tried adding this into Match.swift and instead of ambiguity I get `switch must be exhaustive` with 04/22 snapshot of 5.7 branch.

Reply:
Updated the example.  It's missing a `default: break`.

Reply:
Added `default` and tried with 04/23 snapshot of trunk and the project built successfully, rebased the project with origin/main just in case too.